src/pages/index.js

```diff
@@ -0,0 +1,273 @@
+}
+
+function setProfilePopupListeners() {
+    profileSelectors.editButton.addEventListener("click", () => {
+        resetValidation(profileForm);
+        fillEditFormFields();
+        togglePopup(profileSelectors.editPopup);
+    });
+    
+    profileForm.addEventListener("submit", (evt) => {
+        evt.preventDefault();
+        const popup = profileSelectors.editPopup;
+        const { loadingAnimation, originalText } = submitButtonText(popup);
+        api.editUserInfo(profileSelectors.editName.value, profileSelectors.editDescription.value).
+        then(res => editUserInfo(res.name, res.about)).
+        catch(err => console.log(err)).
+        finally(() => {
+            clearInterval(loadingAnimation);
+            getSubmitButton(popup).textContent = originalText;
+            togglePopup(popup);
+        });
+    });
+}
+
+function setAvatarPopupListeners() {
+    editAvatarSelectors.editAvatarButton.addEventListener("click", () => {
+        resetValidation(editAvatarSelectors.editAvatarPopup);
+        togglePopup(editAvatarSelectors.editAvatarPopup);
+    });
+
+    editAvatarForm.addEventListener("submit", (evt) => {
+        evt.preventDefault();
+        const popup = editAvatarSelectors.editAvatarPopup;
+        const { loadingAnimation, originalText } = submitButtonText(popup);
+        api.editAvatar(editAvatarSelectors.editAvatarLink.value).
+        then(res => editUserInfo(null, null, res.avatar)).
+        catch(err => console.log(err)).
+        finally(() => {
+            clearInterval(loadingAnimation);
+            getSubmitButton(popup).textContent = originalText;
+            editAvatarForm.reset();
+            disableButton(editAvatarForm);
+            togglePopup(popup);
+        });
+    });
+}
+
+function renderCard(data) {
+    cardList.prepend(getCardElement(data));
+}
+
+function openFullImage(data) {
+    fullImageSelectors.fullImageImgElement.src = "";
+    fullImageSelectors.fullImageImgElement.alt = "";
+    fullImageSelectors.fullImageText.textContent = "";
+    togglePopup(fullImageSelectors.fullImagePopup);
+    fullImageSelectors.fullImageImgElement.src = data.src;
+    fullImageSelectors.fullImageImgElement.alt = data.alt;
+    fullImageSelectors.fullImageText.textContent = data.name;
+}
+
+function setDeleteCardPopupListeners(card) {
+    const deleteButton = deleteCardSelectors.deleteCardButtonDelete;
+    const cancelButton = deleteCardSelectors.deleteCardButtonCancel;
+
+    const deleteButtonHandler = () => {
+        return function handler() {
+            const {loadingAnimation, originalText} = submitButtonText(deleteCardSelectors.deleteCardPopup);
+            api.deleteCard(card.id).catch(err => console.log(err)).finally(() => {
+                clearInterval(loadingAnimation);
+                getSubmitButton(deleteCardSelectors.deleteCardPopup).textContent = originalText;
+                togglePopup(deleteCardSelectors.deleteCardPopup);
+                card.remove();
```

Comment:
You need to **delete** cards only in a `then` block after a **successful** response from the server, so that if there is an error, the card should stay in the `DOM`.